### ISSUE TITLE
Just a quick little clarification on how to utilize this mod

### ISSUE THREAD

**Issue by notIkibo** [horizr/one-time-overrides#2]
It seems like the directory "one-time-overrides" is not automatically created and should be in the folder where also the "config" and "mods" directories are (it says .minecraft, but on any other than the standard launcher, it's depending on instance).

But now there are two possibilities and only one makes sense:

When exporting my modpack: 
1. Anything that is in the folder, should not also be in the exported data (e.g. I don't want options.txt to be overwritten so I only export one-time-overrides that has options.txt innit instead of also putting option.txt itself into the overrides)
2. I do have both innit and when the modpack gets updated: During the updating process one-time-overrides steps in and stops overwriting any duplicates (which doesn't make sense, bc how should one-time-overrides ever be able to have a say in the updating process; though it would ofc be just a little easier just exporting everything necessary).

I think it's obvious that I'm not that deep in coding, but I really see the potential of this mod and it's a lifesaver. I always hated when mods overwrite my options I configurated for me. 

I need help :o

**Comment by moritzruth** [comment]
> It seems like the directory "one-time-overrides" is not automatically created and should be in the folder where also the "config" and "mods" directories are (it says .minecraft, but on any other than the standard launcher, it's depending on instance).

This is correct.

> Anything that is in the folder, should not also be in the exported data (e.g. I don't want options.txt to be overwritten so I only export one-time-overrides that has options.txt innit instead of also putting option.txt itself into the overrides)

This is also correct.

You can find [an example of how this mod is to be used in the `visum-pack` repository](https://github.com/horizr/visum-pack/tree/main/src/client).

Please let me know if you have any further questions.

**Comment by notIkibo** [comment]
So:
Some things don't work

The one-time-overrides folder created by me doesn't seem to work properly?

I have a lot of folder in there with configurations. Yet it doesn't really install the configs in the config folder. It just creates the folder without the actual file in there. and then I get something like:


[17:24:42] [Render thread/INFO]: Preloaded theme textures: 73
[17:24:42] [Render thread/INFO]: initialize EXIT, elapsed count 1 avg 1208.84ms
[17:24:42] [Render thread/INFO]: postInitialize EXIT, elapsed count 1 avg 7.51ms
[17:24:42] [Render thread/ERROR]: Unhandled game exception
java.lang.IllegalStateException: Some intrusive holders were not added to registry: [Reference{null=air}]
	at net.minecraft.class_2370.method_40276(class_2370.java:352) ~[client-intermediary.jar:?]
	at net.minecraft.class_2378.method_40292(class_2378.java:404) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.handler$dkb002$fabric-registry-sync-v0$onStart(class_310.java:18486) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:747) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:244) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462) [fabric-loader-0.14.19.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.19.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.19.jar:?]
[17:24:42] [Render thread/INFO]: Stopping!
[17:24:43] [Render thread/ERROR]: Shutdown failure!
java.util.ConcurrentModificationException: null
	at java.util.HashMap.forEach(HashMap.java:1424) ~[?:?]
	at net.minecraft.class_1060.close(class_1060.java:173) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.close(class_310.java:1092) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1490(class_310.java:1068) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:260) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462) [fabric-loader-0.14.19.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.19.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.19.jar:?]
	
at the end and mc crashes. I downloaded my modpack from modrinth and before starting changed the config file
(bc right from the getgo I would get the error shown above) and mc started

**Comment by moritzruth** [comment]
That error message doesn’t seem to have anything to do with this mod. The mod does all of its work before anything Minecraft-related happens. (It would probably even work with any other game using Fabric as its mod loader.)

I think I can’t help you. I just want to point out that I only tested the mod with Minecraft versions `1.1.8.2` and `1.19.2` and other versions may not work.

**Comment by notIkibo** [comment]
Mhm, after that message I tried to find the problematic config. I suspected you (though you mod is just too good to be the problem)
It was a little change in a config I did before exporting it.
https://github.com/quiqueck/BetterEnd/issues/218 

I'll now try again and hope everything works. If anything else happens, I'll happily come back and annoy you

**Comment by notIkibo** [comment]
Since I have two versions now, I downloaded the first version; changed the options.txt file like (Auto Jump on) etc. and updated it to the fixed version. My changed to the options.txt are gone. I saved it all and it's now the version that exists in the one-time-overrides folder.

**Comment by moritzruth** [comment]
> Aka it doesn't work QwQ. Are you gonna work on that? Because it behaves like a normal override folder
The last time I checked, the mod worked as intended. I see no reason for why it suddenly should have stopped working.

How did you apply your update? What launcher are you using? Maybe the launcher is creating a new instance every time you update the pack instead of reusing the old directory.

**Comment by notIkibo** [comment]
I was using the prism Launcher.

My modpack (modrinth.com/modpack/ylc) just hast two versions. 0.5 and 0.6
I had 0.5 (changed some options to see if it works) then updated to 0.6. Same instance. Same everything.
Though the shaders are not in the one-time-overrides in the 0.6 version. Only their configuration (makes more sense that way)

**Comment by notIkibo** [comment]
Exporting was made by ATLauncher.
Consisted of:
config
mods
resource packs
shader packs
one-time-overrides
   config
   shader packs
   options.txt

**Comment by notIkibo** [comment]
I understand now.

Prism Launcher deletes the options.txt file.
I watched while the updating to 0.7 now. I had the folder open. At the end of the update process. options.txt is gone and one-time-overrides obviously gives the options file it has after the game launches.

Imma either look, if there's like a setting to that or/and report that to prisms support
![image](https://github.com/horizr/one-time-overrides/assets/106886638/023eec29-f4a5-40f2-a4e4-046803fbbfcd)

**Comment by notIkibo** [comment]
Here we go again:
I have literally no idea anymore what happens.
This may be not be due to one-time-overrides, but rather any bug of any bug of any mod of any certain circumstance. (So I'm venting)
I updated my modpack to 0.7
I have updated my little resource pack, added ToolTipFix and I apparently had my resource packs in the one-time-overrides, just like my shaders even though I only need the shader configurations innit. So I tidied stuff up and didn't change the options.txt
But: When updating to 0.7. My options get removed and instead I'm greeted with the standard options file. Hooray.
So: Now I'm like "Huh, that's weird. I close the game and just replace it with my options instead."
Now, that's a solution! No. You start the game and you are greeted with the standard options file.

I don't know why. Uncertainty is such a killer rn.

Everything is fine with 0.6, but 0.7 is a whole nother level
